modules/kotest/src/main/kotlin/org/testcontainers/containers/StartablePerTestListener.kt

```diff
@@ -0,0 +1,63 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.testcontainers.lifecycle.Startable
+import org.testcontainers.lifecycle.TestDescription
+import org.testcontainers.lifecycle.TestLifecycleAware
+import java.util.*
+
+
+/**
+ * [StartablePerTestListener] starts the given [startable] before execution of each test in the spec
+ * and stops after execution of each test. If the [startable] also inherit from [TestLifecycleAware]
+ * then its [beforeTest] and [afterTest] method are also called by the listener.
+ *
+ * [startable] can any of [GenericContainer] [DockerComposeContainer] [LocalStackContainer] etc.
+ *
+ * This should be use when you want fresh container for each test.
+ *
+ * @see[StartablePerSpecListener]
+ * */
+internal class StartablePerTestListener(private val startable: Startable) : TestListener {
+
+    override suspend fun beforeTest(testCase: TestCase) {
+        withContext(Dispatchers.IO) {
+            runBeforeTestForTestLifecycleAware(testCase)
+            startable.start()
+        }
+    }
+
+    override suspend fun afterTest(testCase: TestCase, result: TestResult) {
+        withContext(Dispatchers.IO) {
+            runAfterTestForTestLifecycleAware(testCase, result)
+            startable.stop()
+        }
+    }
+
+    private fun runBeforeTestForTestLifecycleAware(testCase: TestCase) {
+        startable.toTestLifecycleAware()?.beforeTest(testCase.toTestDescription())
+    }
+
+    private fun runAfterTestForTestLifecycleAware(testCase: TestCase, result: TestResult) {
+        startable.toTestLifecycleAware()?.afterTest(testCase.toTestDescription(), Optional.ofNullable(result.error))
+    }
+
+    private fun Startable.toTestLifecycleAware() = this as? TestLifecycleAware
+}
+
+
+private fun TestCase.toTestDescription() = object : TestDescription {
+
+    override fun getFilesystemFriendlyName(): String {
+        return name.replace(" ", "-")
```

Comment:
what if `name` contains `/`, `..` or similar characters?

Reply:
Will encoding the name will be ok like in SpockTestDescription?
https://github.com/testcontainers/testcontainers-java/blob/9c4756f73dde77b3df1e4b77aeccaa3de818e935/modules/spock/src/main/groovy/org/testcontainers/spock/SpockTestDescription.groovy#L30-L35